### PR TITLE
Remove endpoint related RBAC from scheduler cluster role

### LINKED ISSUE #117089: remove RBAC rules allowing for creation and update of endpoints by the kubescheduler
### What would you like to be added?

Not added, but removed:

Looking at this comment: https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L551
this should be removed once the migration to lease in leader-election was done. Can this be removed already? I mean, the following 2 lines: https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L551-L553

### Why is this needed?

Security folks internally were asking if we could get rid of this upstream if it is not needed anymore (if this "migration to lease in leader-election"  was fully done)

### PR DESCRIPTION
**Pull request by sathyanarays**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Previously, the kube-scheduler was using endpoint resources for leader election. So, the scheduler cluster role had RBACs to manage the endpoint resources. Now, the leader election is done only using the lease resource. Hence, the corresponding RBACs can be removed.

#### Which issue(s) this PR fixes:

Fixes #117089 

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ialidzhikov** [comment]
@sathyanarays , can you check the failing unit test?

**Comment by alculquicondor** [comment]
`TestBootstrapClusterRoles` needs updating.

lgtm from SIG scheduling otherwise

**Comment by ialidzhikov** [comment]
/cc @wojtek-t 
for approval

File: plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go [hunk 1]
```diff
@@ -548,9 +548,6 @@ func ClusterRoles() []rbacv1.ClusterRole {
 		// TODO: scope this to the kube-system namespace
 		rbacv1helpers.NewRule("create").Groups(coordinationGroup).Resources("leases").RuleOrDie(),
 		rbacv1helpers.NewRule("get", "update").Groups(coordinationGroup).Resources("leases").Names("kube-scheduler").RuleOrDie(),
-		// TODO: Remove once we fully migrate to lease in leader-election.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go:551]: Would a good next step be to log an Error if using anything other than `leases`?

[the remark this answers is not in the document]
Comment (liggitt) [reply plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go:551]: there's two pieces around the migration:
1. what the client-go leader library supports, where we have to consider all possible consumers of the library
2. what the scheduler supports

For this role, we only have to look at what the scheduler supports, and that was limited to only using `leases` in 1.27 in https://github.com/kubernetes/kubernetes/pull/114055, so pruning endpoints out of the kube-scheduler role in 1.28 is fine.

[the remark this answers is not in the document]
Comment (liggitt) [reply plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go:551]: > Would a good next step be to log an Error if using anything other than `leases`?

This PR is fine as-is.

Adding logging/warning to the client-go leader library when using anything other than leases sounds like a good next step in a separate PR.

[the remark this answers is not in the document]
Comment (wojtek-t) [reply plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go:551]: OK - I missed https://github.com/kubernetes/kubernetes/pull/114055
Given that, I agree this PR makes sense.

[the remark this answers is not in the document]
Comment (alculquicondor) [reply plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go:551]: And the validation check is still there https://github.com/kubernetes/kubernetes/blob/ad18954259eae3db51bac2274ed4ca7304b923c4/pkg/scheduler/apis/config/validation/validation.go#L46-L51

**Comment by ialidzhikov** [comment]
I created https://github.com/kubernetes/kubernetes/pull/117414 for similar cleanup on kube-controller-manager side.

**Pull request merged**